### PR TITLE
Implement and tested go-tpm direct function TPM2_GetRandom

### PR DESCRIPTION
**Pull request by matt-tsai**

### CODE REVIEW

File: direct/tpm2/get_random_test.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	defer thetpm.Close()
+
+	grc := GetRandom{
+		BytesRequested: 16,
+	}
+
+	if _, err := grc.Execute(thetpm); err != nil {
+		t.Fatalf("GetRandom failed: %v", err)
+	}
+
```
Comment (chrisfenner) [inline direct/tpm2/get_random_test.go:23]: nit: can delete this blank line

File: direct/tpm2/tpm2.go [hunk 2]
```diff
@@ -355,6 +355,33 @@ type CreateLoadedResponse struct {
...
+// Execute executes the command and returns the response.
+func (cmd *GetRandom) Execute(t transport.TPM, s ...Session) (*GetRandomResponse, error) {
+	var rsp GetRandomResponse
+	if err := execute(t, cmd, &rsp, s...); err != nil {
+		return nil, err
+	}
+	return &rsp, nil
+}
+
+// GetRandomReponse is the reponse from TPM2_GetRandom
```
Comment (chrisfenner) [inline direct/tpm2/tpm2.go:376]: nit: end this sentence in a period

**Review by chrisfenner — APPROVED** [review]
Looks great, thanks for the change! I have two tiny nits.

File: direct/tpm2/tpm2.go [hunk 3]
```diff
@@ -355,6 +355,33 @@ type CreateLoadedResponse struct {
 // Response implements the Response interface.
 func (*CreateLoadedResponse) Response() tpm.CC { return tpm.CCCreateLoaded }
 
+// GetRandom is the input to TPM2_GetRandom.
+// See definition in Part 3, Commands, section 16.1
+type GetRandom struct {
+	BytesRequested uint16
```
Comment (alexmwu) [inline direct/tpm2/tpm2.go:361]: nit: add the description "number of octets to return" from the spec as a comment

File: direct/tpm2/get_random_test.go [hunk 4]
```diff
@@ -0,0 +1,24 @@
+package tpm2
```
Comment (alexmwu) [inline direct/tpm2/get_random_test.go:1]: (not a blocking comment) how do we want to organize tests? having a test per command may make the directory quite large. we could have a test directory like in `tpm2` or we could split by use case.

**Review by alexmwu — APPROVED** [review]

File: direct/tpm2/get_random_test.go (hunk 4 above)
Comment (chrisfenner) [reply direct/tpm2/get_random_test.go:1]: My mild opinion is that I like having more and smaller test files. A test directory with e.g., one file per command tested would be groovy.

Comment (alexmwu) [reply direct/tpm2/get_random_test.go:1]: I generally agree this makes tests easier to find, and we should go for this. I do wonder about tests that use multiple commands like a potential policy auth test with multiple policies.

Comment (chrisfenner) [reply direct/tpm2/get_random_test.go:1]: Right, and we'd have a lot of helpers too. So we might have some per-command tests as well as some "integration-y" tests that really cover a bunch of commands. That's a good reason not to be rigid on "one file per command" but in general I think it's ok to have a lot of files.

**Review by chrisfenner — APPROVED** [review]

**Comment by chrisfenner** [comment]
LGTM - Matt can you update your branch with a rebase?

**Pull request merged**